### DISCUSSION TITLE
Ordering levels for indicators

### DISCUSSION THREAD

**Discussion by ukraine-sdg** [open-sdg/open-sdg#1467] [Q&A]
I am using open-sdg to create the platform for Ukraine. 

Could you, please, explaine how can I organise my data to have the orgered levels for subcategories?

**Comment by brockfanning** [comment]
@ukraine-sdg The good news is that Open SDG should do this automatically, when your data calls for it. So it really depends on your data. Open SDG will look at your data and decide whether it has parent/child relationships (also known as "edges").

For example, the following data would not produce any edges:

Year | Sex | Age | Value
--- | --- | --- | ---
2020 | Female | | 100
2020 | Male | | 50
2020 | | Under 5 years | 75
2020 | | 5 years and older | 80

But by contrast, the following data would produce an "edge", where the "Sex" column is the parent of the "Age" column:

Year | Sex | Age | Value
--- | --- | --- | ---
2020 | Female | | 100
2020 | Male | | 50
2020 | Female | Under 5 years | 75
2020 | Female | 5 years and older | 80
2020 | Male | Under 5 years | 30
2020 | Male | 5 years and older | 40

The reason is that in the first example, "Sex" and "Age" are completely independent of each other. But in the second example, you cannot disaggregate by "Age" without also disaggregating by "Sex". This makes "Sex" the parent and "Age" the child. In practice, on the platform, "Age" will appear below "Sex" and indented to the right, to indicate that it's a child field.

**Reply by ukraine-sdg** [reply]
Thank you, Brock.

I also have a question about the orders of the levels for each group.
For example, I want to have in the menu for categories the levels for sex in this order: male, female.
How can I do this?

**Comment by brockfanning** [comment]
@ukraine-sdg There are 3 different ways to handle the order of the disaggregations and disaggregation values:

1. Alphabetical order - this is the way it will be ordered if you don't use either of the other two approaches. To use this approach you don't have to do anything special. The dissaggregations and disaggregation values will be sorted alphabetically.
2. Ordered by position in the data: Disaggregations and disaggregation values will be ordered according to the order in which they appear in the source data (eg, the CSV file). To use this approach you need to have this in your data configuration (eg, config_data.yml):

        datapackage:
          sorting: default

3. Ordered by a separate "data schema" file. This is the most technically involved, but allows full control over the order of all the disaggregations and disaggregation values. To use this approach you have create data schema files. More information on this is here: More information on that is here: https://open-sdg.readthedocs.io/en/latest/data-configuration/#data_schema
